packages/nouns-contracts/contracts/StreamEscrow.sol

```diff
@@ -0,0 +1,307 @@
+        Stream memory stream = streams[nounId];
+        ethStreamedPerTick -= stream.ethPerTick;
+        ethStreamEndingAtTick[stream.lastTick] -= stream.ethPerTick;
+
+        // calculate how much needs to be refunded
+        uint256 ticksLeft = stream.lastTick - currentTick;
+        uint256 amountToRefund = stream.ethPerTick * ticksLeft;
+        (bool sent, ) = msg.sender.call{ value: amountToRefund }('');
+        require(sent, 'failed to send eth');
+
+        emit StreamCanceled(nounId, amountToRefund);
+    }
+
+    /**
+     * @notice Fast-forwards a stream by a certain number of ticks.
+     * @param nounId The ID of the Noun token to fast-forward the stream for.
+     * @param ticksToForward The number of ticks to fast-forward the stream by.
+     * @custom:throws 'ticksToForward must be positive' if `ticksToForward` is not positive.
+     * @custom:throws 'ticksToFoward too large' if `ticksToForward` is larger than the remaining ticks.
+     * @custom:throws 'not noun owner' if the caller is not the Noun owner.
+     * @custom:throws 'stream not active' if the stream is not active.
+     */
+    function fastForwardStream(uint256 nounId, uint32 ticksToForward) public {
+        require(ticksToForward > 0, 'ticksToForward must be positive');
+        require(nounsToken.ownerOf(nounId) == msg.sender, 'not noun owner');
+
+        Stream memory stream = streams[nounId];
+        uint32 currentTick_ = currentTick;
+        require(isStreamActive(stream, currentTick_), 'stream not active');
+        uint32 lastTick = stream.lastTick;
+
+        // move last tick
+        require(ticksToForward <= lastTick - currentTick_, 'ticksToFoward too large');
+        uint32 newLastTick = lastTick - ticksToForward;
+
+        streams[nounId].lastTick = newLastTick;
+        ethStreamEndingAtTick[lastTick] -= stream.ethPerTick;
+
+        if (newLastTick > currentTick_) {
+            // stream is still active, so register the new end tick
+            ethStreamEndingAtTick[newLastTick] += stream.ethPerTick;
+        } else {
+            // no more ticks left, so finished the stream
+            ethStreamedPerTick -= stream.ethPerTick;
+        }
+
+        uint256 ethToStream = ticksToForward * stream.ethPerTick;
+        sendETHToTreasury(ethToStream);
+
+        emit StreamFastForwarded(nounId, ticksToForward, newLastTick);
+    }
+
+    /**
+     * @notice Checks if a stream is active (not canceled and not finished) for a specific Noun token.
+     * @param nounId The ID of the Noun token to check the stream for.
+     * @return `true` if the stream is active, `false` otherwise.
+     */
+    function isStreamActive(uint256 nounId) public view returns (bool) {
+        return !streams[nounId].canceled && streams[nounId].lastTick > currentTick;
+    }
+
+    function isStreamActive(Stream memory stream, uint32 tick) internal pure returns (bool) {
+        return !stream.canceled && stream.lastTick > tick;
+    }
+
+    modifier onlyDAO() {
```

Comment:
nit: official solidity style guides say modifiers go higher in the file, in the area of state vars, wdyt?

Reply:
sure 9a301cd161c0772c278f12eb5c97f222f7cf8e3e